docs/index.rst

```diff
@@ -54,3 +54,49 @@ The template currently implements the following optional flags, all of which def
 * ``include_example_code``: This option will fill your new package with some example functions to allow you to test it.
 * ``use_compiled_extensions``: This turns on the features needed to support compiled extensions as described in :ref:`extensions`.
 * ``enable_dynamic_dev_versions``: This enables a feature which ensures that ``my_package.__version__`` always returns the current git version as calculated by ``setuptools_scm`` when the package is installed as an editable install. See :ref:`dev-versions` for more details.
+* ``include_cruft_update_github_repo``: This option adds a github workflow with pulls in the latest changes from the template every Monday morning and creates a PR against the repo which can then be accepted or closed.
+* ``use_extended_ruff_linting``: This option flag enables the stricter ruff rules. Recommend `Y` on creation of a new project.
+
+Pre-commit
+==========
+
+Pre-commit is configured through ``.pre-commit-config.yaml`` and can be installed locally and ran:
+
+.. code-block:: bash
+
+    $ pre-commit run --all-files
+
+It also possible to use the tox environment to run it and and is integrated into the CI.
+
+Within ``.pre-commit-config.yaml``, there are several tools and each one is configured either within the ``.pre-commit-config.yaml`` or for larger tools like ruff, it is has a dedicated config file ``.ruff.toml`` and we strongly recommend using the full set of rules when you setup your package.
+
+Updating to a new version of the template
+=========================================
+
+It will be simplest to updating a package to a newer template by waiting for the GitHub workflow to trigger.
+This will trigger a pull request one can review and merge via GitHub's UI.
+
+If you do not want to wait, or want to do it manually, you will have to use cruft's CLI.
+For this you will need to install cruft locally and then you can check the status of the package by running:
+
+.. code-block:: console
+
+    $ cruft check
+
+This will let you know whether the repository is up to date or not.
+From there, you can run:
+
+.. code-block:: console
+
+    $ cruft update
+
+To update the repo and if there is a case conflicting files, ``.rej`` files will be created and you will have to manually deal with them and merge changes.
+
+If you need up update one of the variables in the package ``.cruft.json`` for example, changing a ``n`` to a ``` this can be done using:
```

Comment:
```suggestion
If you need up update one of the variables in the package ``.cruft.json`` for example, changing a ``n`` to a ``y`` this can be done using:
```